### PR TITLE
test(components/google-cloud): Unit test for remote_runner module

### PR DESCRIPTION
**Pull request by SinaChavoshi**
Adding unit tests for remote_runner.py

### CODE REVIEW

File: components/google-cloud/tests/aiplatform/test_remote_runner.py [hunk 1]
```diff
@@ -13,10 +13,272 @@
...
+        result = remote_runner.make_output(output_object)
+        self.assertEqual(result, expected_result)
+
+    @mock.patch.object(
+        utils, "is_mb_sdk_resource_noun_type", autospec=True, return_value=True
+    )
+    def test_make_output_with_resouce_name_returns_resouce_name_value(
+        self, mock_utils
+    ):
+        output_object = mock.Mock()
```
Comment (sasha-gitg) [inline components/google-cloud/tests/aiplatform/test_remote_runner.py:186]: If you wanted to test this without mocking:

```python
output_object = aiplatform.Model._empty_constructor(project='test-project')
output_object._gca_resource = aiplatform.gapic.Model(name='resource_name')
```

**Review by sasha-gitg — APPROVED** [review]
LGTM. Thanks Sina!

File: components/google-cloud/tests/aiplatform/test_remote_runner.py [hunk 2]
```diff
@@ -13,10 +13,272 @@
...
+                "artifacts": {},
+                "outputFile": "/gcs/test_path/executor_output.json"
+            }
+        }, "test_resource_uri")
+
+        mock_open.return_value.__enter__().write.assert_called_once_with(
+            json.dumps({})
+        )
+
+    def test_resolve_input_args_non_resouce_noun_removes_gcs_prefix(self):
```
[the remark this answers is not in the document]
Comment (SinaChavoshi) [reply components/google-cloud/tests/aiplatform/test_remote_runner.py]: good catch! done.

File: components/google-cloud/tests/aiplatform/test_remote_runner.py [hunk 3]
```diff
@@ -13,10 +13,272 @@
...
+        type_to_resolve = aiplatform.base.AiPlatformResourceNoun
```
[the remark this answers is not in the document]
Comment (SinaChavoshi) [reply components/google-cloud/tests/aiplatform/test_remote_runner.py:127]: good point, updated accordingly.

File: components/google-cloud/tests/aiplatform/test_remote_runner.py (hunk 1 above)
Comment (SinaChavoshi) [reply components/google-cloud/tests/aiplatform/test_remote_runner.py:186]: Done! nice I like using the actual method in this case over mock object, thank you.

File: components/google-cloud/tests/aiplatform/test_remote_runner.py [hunk 4]
```diff
@@ -13,10 +13,272 @@
...
+
+    def test_resolve_init_args_key_ends_with_name_removes_gcs_prefix(self):
+        key = 'resouce_name'
+        value = '/gcs/test_resouce_name'
+        expected_result = 'test_resouce_name'
+
+        result = remote_runner.resolve_init_args(key, value)
+        self.assertEqual(result, expected_result)
+
+    @mock.patch.object(
```
[the remark this answers is not in the document]
Comment (SinaChavoshi) [reply components/google-cloud/tests/aiplatform/test_remote_runner.py:166]: good catch! removed it.

**Review by sasha-gitg — APPROVED** [review]

**Pull request merged**